File: festival-web/src/main/java/cn/bdqfork/web/service/JsonMessageCodec.java
package cn.bdqfork.web.service;

import io.vertx.core.buffer.Buffer;
import io.vertx.core.eventbus.MessageCodec;
import io.vertx.core.json.Json;

/**
 * @author bdq
 * @since 2020/1/26
 */
public class JsonMessageCodec implements MessageCodec<Object, Object> {
    public static final String NAME = "json";

    @Override
    public void encodeToWire(Buffer buffer, Object o) {
        buffer.appendBuffer(Json.encodeToBuffer(o));
    }

    @Override
    public Object decodeFromWire(int pos, Buffer buffer) {
        return Json.decodeValue(buffer);
    }

    @Override
    public Object transform(Object o) {
        return o;
    }

    @Override
    public String name() {
        return NAME;
    }

    @Override
    public byte systemCodecID() {
        return -1;
    }
}

File: festival-kotlin-web/src/main/kotlin/cn/bdqfork/kotlin/web/service/JsonMessageCodec.kt
package cn.bdqfork.kotlin.web.service

import io.vertx.core.buffer.Buffer
import io.vertx.core.eventbus.MessageCodec
import io.vertx.core.json.Json

/**
 * @author bdq
 * @since 2020/1/26
 */
class JsonMessageCodec : MessageCodec<Any?, Any?> {
    override fun encodeToWire(buffer: Buffer, o: Any?) {
        buffer.appendBuffer(Json.encodeToBuffer(o))
    }

    override fun decodeFromWire(pos: Int, buffer: Buffer): Any? {
        return Json.decodeValue(buffer)
    }

    override fun transform(o: Any?): Any? {
        return o
    }

    override fun name(): String {
        return NAME
    }

    override fun systemCodecID(): Byte {
        return -1
    }

    companion object {
        const val NAME = "json"
    }
}